test/unit/apicast/provider_source_test.rb

```diff
@@ -32,7 +32,7 @@ def test_services
 
     service.stubs(updated_at: Time.now)
     service_attributes =  @source.attributes_for_proxy['services'][0]
-    assert_equal service.updated_at, service_attributes['updated_at']
+    assert_equal service.updated_at.iso8601, service_attributes['updated_at']
```

Comment:
Hmmm, why this?...

Reply:
Here's the commit: https://github.com/3scale/porta/commit/0a8faa77c583bc0e68b58a20bc7e68b29139c656

I tried the test and it fails if I remove that conversion to `iso8601`. It seems Rails' `as_json` changed the way it returns dates and that made the tests fail.

Reply:
Yeah, what I mean is - if the test is failing now, does it mean that the application behavior has changed?? Maybe it means we need to fix something else??

Reply:
good point

Reply:
Let me check

Reply:
Hm.... So, I don't really understand this, on line `:33` there is `service.stubs(updated_at: Time.now)` and I think this is actually what causes the issue.

I think it's because `updated_at` is `ActiveSupport::TimeWithZone` and it handles the equality with ISO8601 correctly, while `Time` doesn't.

But actually, I've dug into this class, and I think it's a deprecated functionality, which probably needs to be removed, so I am not bothered by it anymore :) 
In fact, I think I'll create a JIRA to remove the related classes.

Reply:
Sorry, I was wrong, the endpoint that uses this class is still in use... 
But well, I am not sure to be honest whether this specific field `updated_at` is actually useful.

Reply:
I think it is much better to compare a string to a string instead of relying on implicit coercion. And we make sure that the actual format of the string is preserved across Rails and other libraries upgrades.